src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3101,5 +3101,218 @@ internal virtual ImmutableArray<LabelSymbol> Labels
                 return ImmutableArray<LabelSymbol>.Empty;
             }
         }
+
+        /// <summary>
+        /// Perform a lookup for the specified method on the specified type.  Perform overload resolution
+        /// on the lookup results.
+        /// </summary>
+        /// <param name="patternType">Type to search.</param>
+        /// <param name="methodName">Method to search for.</param>
+        /// <param name="lookupResult">Passed in for reusability.</param>
+        /// <param name="warningsOnly">True if failures should result in warnings; false if they should result in errors.</param>
+        /// <param name="diagnostics">Populated with binding diagnostics.</param>
+        /// <returns>The desired method or null.</returns>
+        internal MethodSymbol FindPatternMethod(TypeSymbol patternType, string methodName, LookupResult lookupResult,
+                                                SyntaxNode syntaxExpr, bool warningsOnly, DiagnosticBag diagnostics,
+                                                SyntaxTree syntaxTree)
+        {
+            Debug.Assert(lookupResult.IsClear);
+
+            MessageID messageID;
+            switch (methodName)
+            {
+                case WellKnownMemberNames.GetDisposeMethodName:
```

Comment:
I think this should just be called `DisposeMethodName` since the pattern seems to be to use the exact name of the method.